### ISSUE TITLE
TypedDict cannot be used in dict.update

### ISSUE THREAD

**Issue by ojii** [astral-sh/ty#2853]
### Summary

```python
from typing import TypedDict

class Foo(TypedDict):
    a: str
    
data: dict[str, str] = {"b": "2"}
foo: Foo = {"a": "1"}
data.update(foo) # No overload of bound method `update` matches arguments (no-matching-overload)
data.update(iter(foo)) # this type checks
```

I would assume that `TypedDict` can be used in `dict.update`, am I missing something?

### Version

ty 0.0.17 (8cec85718 2026-02-13)

**Comment by sharkdp** [comment]
> I would assume that `TypedDict` can be used in `dict.update`, am I missing something?

`data` is of type `dict[str, str]`, so a type checker should maintain the property that all of its values are of type `str`. The `Foo` TypedDict specifies that its `"a"` key maps to a value of type `str` (check), but there could be extra items:
```py
class FooExtended(Foo):
    other: int

foo_e: FooExtended = {"a": "1", "other": 123}

def update_with(foo: Foo):
    data: dict[str, str] = {"b": "2"}
    data.update(foo)

update_with(foo_e)  # oops! now `data` contains a value of type `int` as well
```

So a type checker should not allow an `data.update(foo)` call with `foo` of type `Foo`. However, I *think* we should allow `data.update(foo)` if `Foo` is a *closed* TypedDict, or a TypedDict with `extra_items=str`. But we do not support [PEP 728](https://peps.python.org/pep-0728) yet (it is targeted for Python 3.15). We have https://github.com/astral-sh/ty/issues/154 to track this.

**Comment by ojii** [comment]
That makes perfect sense, thank you very much for the explanation. Changing data to `data: dict[Any, Any]` also makes it type check.
